es6-shim.js

```diff
@@ -291,6 +291,13 @@
       return x != null && Object(x) === x;
     },
 
+    TypeIsNotObject: function (val) {
+      // This is faster when TypeIsObject may commonly return false.
+      return val === void 0 || val === null || val === true || val === false ||
```

Comment:
`val == null` is actually slower than an explicit test against `null` and `undefined`, when you look at v8's optimized assembly output.  The form shown here was pretty extensively benchmarked.  It might look a little weird, but it's pretty straightforward to read (a plus) and happens to be the fastest option.